### PR TITLE
http headers on sample

### PR DESCRIPTION
**Pull request by Noly**
Url API now can retrieve response header value attached on each generated sample included a response error.
Setting up `return_headers` to true

### CODE REVIEW

File: internal/inputs/http_test.go [hunk 1]
```diff
@@ -7,102 +7,318 @@ package inputs
...
+						URL:           "/todos",
+						Timeout:       5100,
+						ReturnHeaders: true,
+					},
+				},
+			},
+			[]interface{}{
+				map[string]interface{}{
+					"error":                     "Missing Required Parameters",
+					"api.StatusCode":            200,
```
Comment (ardias) [inline internal/inputs/http_test.go:217]: this is strange. If statusCode is 200, "error" should not be present. I think error is only when the http call fails, no?

**Comment by ardias** [comment]
Looks good. Check the windows test please since they seem to be failing

File: internal/inputs/http_test.go (hunk 1 above)
Comment (Noly) [reply internal/inputs/http_test.go:217]: Yes, totally. As an expected value could be more real with a 5xx status code. Will change it.

However I've saw apis that return success status code with error message in body XD

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## 1.3.1 - 2020-05-XX
+### Added
+- Url API now can retrieve response header value attached on each generated sample included a response error.
```
Comment (ardias) [inline CHANGELOG.md:10]: I would rephrase as :
Url API can now include all the response headers in the resulting sample. Set `return_headers` to true to enable.

File: docs/apis/url.md [hunk 3]
```diff
@@ -105,4 +106,18 @@ apis:
             keys: [net.connectionsReading, net.connectionsWriting, net.connectionsWaiting]
     math:
       net.connectionsDroppedPerSecond: ${net.connectionsAcceptedPerSecond} - ${net.handledPerSecond}
-```
\ No newline at end of file
+```
+
+##  <a name='ReturnResponseHeaders'></a>Include response headers on sample
+
+To include response headers on the metric sample setting `return_headers` attribute to true is needed.
```
Comment (ardias) [inline docs/apis/url.md:113]: "To include response headers on the metric sample set `return_headers` attribute to true."

File: internal/inputs/http.go [hunk 4]
```diff
@@ -246,50 +246,40 @@ func setRequestOptions(request *gorequest.SuperAgent, yml load.Config, api load.
...
-			// load.StoreAppend(theSample)
-			*dataStore = append(*dataStore, theSample)
+	var cb responseBody
 
-			if theSample["error"] != nil && fmt.Sprintf("%v", theSample["error"]) != "false" {
-				load.Logrus.Debugf("http: request failed %v", theSample["error"])
-			}
+	switch t := b.(type) {
+	case []interface{}:
+		cb = newObjectBody(t)
```
Comment (ardias) [inline internal/inputs/http.go:262]: shouldn't this be the other way around? In this case the type `t` is a array on "objects" but you are using 
`newObjectBody` instead of `newArrayBody()`?

**Comment by ardias** [comment]
I'm not completely sure if the new code is equivalent as the old one.

File: internal/inputs/http.go (hunk 4 above)
Comment (Kav91) [reply internal/inputs/http.go:262]: looks like the functions just needs to be renamed?

File: CHANGELOG.md (hunk 2 above)
Comment (Noly) [reply CHANGELOG.md:10]: 👍

File: docs/apis/url.md (hunk 3 above)
Comment (Noly) [reply docs/apis/url.md:113]: 👍

File: internal/inputs/http.go (hunk 4 above)
Comment (Noly) [reply internal/inputs/http.go:262]: Yes, indeed. Thanks

**Comment by Noly** [comment]
> I'm not completely sure if the new code is equivalent as the old one. We may need more robust integration tests for HTTP

Refactor was made after tests were green, manual testing were success. 

Which use case would you like to add as integration test to be more confident with this changes?

**Comment by ardias** [comment]
Tested with **examples/flexConfigs/http-json-example.yml** and the output is the same as the latest release, so I think we're good
